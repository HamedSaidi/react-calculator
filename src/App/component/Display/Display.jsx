import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './Display.css';

export default class Display extends React.Component {
  static propTypes = {
      value: PropTypes.string.isRequired,
  }

  render() {
      const { value } = this.props;
      const classes = cx('component-display', {
          overflow: value.length > 60,
      });

      return (
          <div className={classes}>
              {value}
          </div>
      );
  }
}
