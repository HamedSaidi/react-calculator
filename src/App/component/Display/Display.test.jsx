import React from 'react';
import { shallow } from 'enzyme';

import Display from '.';

describe('Display', () => {
    let consoleSpy;

    beforeAll(() => {
        consoleSpy = jest.spyOn(console, 'error');
    });

    it('renders the Display component and matches snapshot', () => {
        const component = shallow(<Display value="0" />);

        expect(component).toMatchSnapshot();
    });

    it('renders the Display component with overflow class when value length exceeds 60', () => {
        const value = '1'.repeat(61);
        const component = shallow(<Display value={value} />);

        expect(component.find('.overflow')).toHaveLength(1);
    });

    it('shows no errors on console', () => {
        expect(consoleSpy).not.toHaveBeenCalled();
    });
});
