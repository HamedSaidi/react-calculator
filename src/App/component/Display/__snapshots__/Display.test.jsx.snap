// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders the Display component and matches snapshot 1`] = `
<div
  className="component-display"
>
  0
</div>
`;
